ngraph/core/src/op/one_hot.cpp

```diff
@@ -155,12 +158,16 @@ bool op::v1::OneHot::evaluate(const HostTensorVector& output_values, const HostT
     const auto out_shape = out_Pshape.get_shape();
     const size_t axis = get_axis();
     NGRAPH_CHECK(axis >= 0 && axis < out_shape.size(), "Invalid axis value.");
-    const auto depth = get_constant_from_source(input_value(1))->cast_vector<int64_t>()[0];
-    const auto ind_shape = ind_Pshape.get_shape();
-    NGRAPH_CHECK(shape_size(ind_shape) * depth == shape_size(out_shape),
-                 "Incompatible I/O shapes or wrong depth value.");
-    NGRAPH_CHECK(static_cast<int64_t>(out_shape[axis]) == depth, "Incompatible axis and depth values.");
-    return one_hot::evaluate_onehot(output_values, input_values, axis);
+    const auto depth = get_constant_from_source(input_value(1));
+    if (depth) {
+        const auto ind_shape = ind_Pshape.get_shape();
+        const auto depthValue = depth->cast_vector<int64_t>()[0];
+        NGRAPH_CHECK(shape_size(ind_shape) * depthValue == shape_size(out_shape),
+                     "Incompatible I/O shapes or wrong depth value.");
+        NGRAPH_CHECK(static_cast<int64_t>(out_shape[axis]) == depthValue, "Incompatible axis and depth values.");
+        return one_hot::evaluate_onehot(output_values, input_values, axis);
+    }
+    return false;
 }
```

Comment:
this is not right way to put it
_depth_ should be taken not from the constant but from the _input_values_ incoming as _HostTensor_. 
I'm not sure if you would like to handle it here yourself -- but basically it could be done like that:

```suggestion
    const auto depth = std::make_shared<op::v0::Constant>(input_values[1])->cast_vector<int64_t>()[0];
    const auto ind_shape = ind_Pshape.get_shape();
    NGRAPH_CHECK(shape_size(ind_shape) * depth == shape_size(out_shape),
                 "Incompatible I/O shapes or wrong depth value.");
    NGRAPH_CHECK(static_cast<int64_t>(out_shape[axis]) == depth, "Incompatible axis and depth values.");
    return one_hot::evaluate_onehot(output_values, input_values, axis);
```